extensions/interactions/InteractiveMap/InteractiveMapRulesServiceSpec.js

```diff
@@ -0,0 +1,189 @@
+    expect(imrs.Within(EQUATOR_ORIGIN, {
+      p: [0, -90],
+      d: QUARTER_CIRCUMFERENCE - DELTA
+    })).toBe(false);
+    expect(imrs.Within(EQUATOR_ORIGIN, {
+      p: [0, -180],
+      d: HALF_CIRCUMFERENCE - DELTA
+    })).toBe(false);
+    expect(imrs.Within(NORTH_POLE, {
+      p: [90, 180],
+      d: 0.1
+    })).toBe(true);
+    expect(imrs.Within(NORTH_POLE, {
+      p: [90, -180],
+      d: 0.1
+    })).toBe(true);
+    expect(imrs.Within(SOUTH_POLE, {
+      p: [-90, -180],
+      d: 0.1
+    })).toBe(true);
+    expect(imrs.Within(SOUTH_POLE, {
+      p: [-90, 180],
+      d: 0.1
+    })).toBe(true);
+    expect(imrs.Within([-37, -97], {
+      p: [55, -45],
+      d: 11370
+    })).toBe(true);
+    expect(imrs.Within([47, -17], {
+      p: [-81, 117],
+      d: 15890
+    })).toBe(true);
+    expect(imrs.Within([16, -142], {
+      p: [-42, 3],
+      d: 15500
+    })).toBe(false);
+    expect(imrs.Within([83, -127], {
+      p: [27, -123],
+      d: 6220
+    })).toBe(false);
+  });
+
+  it('should have a correct \'not within\' rule', function() {
+    expect(imrs.NotWithin(EQUATOR_ORIGIN, {
+      p: [0, 180],
+      d: HALF_CIRCUMFERENCE + DELTA
+    })).toBe(false);
+    expect(imrs.NotWithin(EQUATOR_ORIGIN, {
+      p: [0, 90],
+      d: QUARTER_CIRCUMFERENCE + DELTA
+    })).toBe(false);
+    expect(imrs.NotWithin(EQUATOR_ORIGIN, {
+      p: [0, -90],
+      d: QUARTER_CIRCUMFERENCE + DELTA
+    })).toBe(false);
+    expect(imrs.NotWithin(EQUATOR_ORIGIN, {
+      p: [0, -180],
+      d: HALF_CIRCUMFERENCE + DELTA
+    })).toBe(false);
+    expect(imrs.NotWithin(EQUATOR_ORIGIN, {
+      p: [0, 180],
+      d: HALF_CIRCUMFERENCE - DELTA
+    })).toBe(true);
+    expect(imrs.NotWithin(EQUATOR_ORIGIN, {
+      p: [0, 90],
+      d: QUARTER_CIRCUMFERENCE - DELTA
+    })).toBe(true);
+    expect(imrs.NotWithin(EQUATOR_ORIGIN, {
+      p: [0, -90],
+      d: QUARTER_CIRCUMFERENCE - DELTA
+    })).toBe(true);
+    expect(imrs.NotWithin(EQUATOR_ORIGIN, {
+      p: [0, -180],
+      d: HALF_CIRCUMFERENCE - DELTA
+    })).toBe(true);
+    expect(imrs.NotWithin(NORTH_POLE, {
+      p: [90, 180],
+      d: 0.1
+    })).toBe(false);
+    expect(imrs.NotWithin(NORTH_POLE, {
+      p: [90, -180],
+      d: 0.1
+    })).toBe(false);
+    expect(imrs.NotWithin(SOUTH_POLE, {
+      p: [-90, -180],
+      d: 0.1
+    })).toBe(false);
+    expect(imrs.NotWithin(SOUTH_POLE, {
+      p: [-90, 180],
+      d: 0.1
+    })).toBe(false);
+    expect(imrs.NotWithin([-37, -97], {
+      p: [55, -45],
+      d: 11370
+    })).toBe(false);
+    expect(imrs.NotWithin([47, -17], {
+      p: [-81, 117],
+      d: 15890
+    })).toBe(false);
+    expect(imrs.NotWithin([16, -142], {
+      p: [-42, 3],
+      d: 15500
+    })).toBe(true);
+    expect(imrs.NotWithin([83, -127], {
+      p: [27, -123],
+      d: 6220
+    })).toBe(true);
+  });
+
+  it('should have a correct fuzzy rule', function() {
+    expect(imrs.FuzzyMatches([0, 0], {
```

Comment:
Question for @BenHenning -- note that the outputs of these are boolean, and not real numbers in [0, 1]. Is it a good idea to do things this way, or should we remove this rule in the interim and add it later when we have the fuzzy rule infrastructure? I think it's weird that this is a fuzzy 'training data' rule but that it seems to have the same structure as the other rules.